recipes/spz/meta.yaml

```diff
@@ -0,0 +1,58 @@
+{% set version = "v2.0.0" %}
+
+package:
+  name: spz
+  version: {{ version }}
+
+source:
+  url: https://github.com/nianticlabs/spz/archive/{{ version }}.tar.gz
+  sha256: fa14626b531c1dee2c29c29b43de8eaa8b773c3b0ef434ee8372c9712dfb6773
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('spz', max_pin='x.x.x') }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ stdlib("c") }}
+    - cmake
+    - ninja
+    - zlib
+
+outputs:
+  - name: spz
+    test:
+      commands:
+        - test -f ${PREFIX}/include/load-spz.h
+        - conda inspect linkages -p $PREFIX spz  # [not win]
+
+  - name: spz-static
+    script: build_static.sh  # [unix]
+    script: bld.bat  # [win]
```

Comment:
This would build a shared version, right?

Reply:
There is no shared version available to build with their stock release. I hope to upstream patches to do so and fix the [include path pollution](https://github.com/nianticlabs/spz/issues/43) at some point as well.

Reply:
I think we could easily patch `CmakeLists.txt` to build a share library.  I believe if you just do something like `sed -i 's/add_library(spz ${spz_sources})/add_library(spz SHARED ${spz_sources})/' CMakeLists.txt` that might be enough to convince it to build a shared library.  

@xhochy in general I think we might want to eventually have some documentation for how to "persuade" CMake and Meson to build shared libraries - I think in most cases a simple `sed` edit is enough.  Autotools and manually created Makefiles can vary a lot in difficulty but some pointers could also be given for those too.

Reply:
While it's easy enough to tell CMake to build `SHARED` with a small patch on unix builds, it's a very different thing on Windows which requires going through and decorating the required classes and functions with `__declspec(dllexport)` to export them. Maintaining *that* set of patches in a packaging system is a giant hassle.

The SPZ project, format, and library is new but has a lot of momentum behind it. Having it in Conda Forge will make it more convenient for people to build with it (and for us to use it in `pdal`, where we can unvendor it).

Reply:
On Windows, you can use `cmake -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON` to export everything.

Reply:
I updated the recipe to be a single `spz` recipe that emits a shared library for both windows and unix. `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON` is a 🔨 that could potentially have some awful consequences if the library authors aren't careful about their variable naming and such, but I we can live with it until we get proper exports upstreamed.

Reply:
@hobu I will say from the many cases we've used that, we have yet to encounter any issues so far.  I think things could get messy if authors aren't careful about duplicate symbol names, but when that happens GCC and Clang nowadays tend to throw errors as well.